android/src/main/java/com/facebook/reactnative/androidsdk/Utility.java

```diff
@@ -100,6 +101,69 @@ public static WritableMap accessTokenToReactMap(AccessToken accessToken) {
         return map;
     }
 
+    public static WritableMap profileToReactMap(Profile profile) {
+        WritableMap map = Arguments.createMap();
+
+        String firstName = profile.getFirstName();
+        if (firstName == null) {
+            map.putNull("firstName");
+        } else {
+            map.putString("firstName", firstName);
+        }
+
+        String lastName = profile.getLastName();
+        if (lastName == null) {
+            map.putNull("lastName");
+        } else {
+            map.putString("lastName", lastName);
+        }
+
+        String middleName = profile.getMiddleName();
+        if (middleName == null) {
+            map.putNull("middleName");
+        } else {
+            map.putString("middleName", middleName);
+        }
+
+        Uri profilePictureUri = profile.getProfilePictureUri(1, 1);
+        if (profilePictureUri == null) {
+            map.putNull("imageURL");
+        } else {
+            // NOTE: Removing the width and height query params from url in order to retrieve the default sized image from Facebook.
+            // Maybe it can be implemented a separated method for retrieve a profile image using width and height params in the future.
+            // But this is not in the scope of the currentProfile method.
+            String urlStringImage = profilePictureUri.toString();
+            map.putString("imageURL", urlStringImage.replace("height=1&width=1&", ""));
+        }
+
+        Uri linkUri = profile.getLinkUri();
+        if (linkUri == null) {
+            map.putNull("linkURL");
+        } else {
+            map.putString("linkURL", linkUri.toString());
+        }
+
+        String userId = profile.getId();
+        if (userId == null) {
+            map.putNull("userID");
+        } else {
+            map.putString("userID", userId);
+        }
+
+        // NOTE: Email doesn't get retrieved directly by the profile instance after a login. You have to fetch the graph API in
+        // order to get this value.
+        map.putNull("email");
```

Comment:
why do we need to place email here if it doesn't come from the endpoint itself?

Reply:
Because the Profile class inside the Java SDK doesn't expose the email field. I place email=null for maintain the same behaviour with iOS. Do you think that is more correct to move this concept in JS?